zeebe/exporters/camunda-exporter/src/test/java/io/camunda/exporter/CamundaExporterIT.java

```diff
@@ -254,6 +256,70 @@ void shouldNotErrorIfOldExporterRestartsWhileNewExporterHasAlreadyStarted(
         .isTrue();
   }
 
+  @TestTemplate
+  void shouldCreateHarmonizedSchemaEagerlyOnOpen(
+      final ExporterConfiguration config, final SearchClientAdapter ignored) {
+    // given
+    final CamundaExporter camundaExporter = new CamundaExporter();
+    camundaExporter.configure(getContextFromConfig(config));
+
+    final var adapter = ClientAdapter.of(config);
+    final var mappingsBeforeOpen =
+        adapter.getSearchEngineClient().getMappings(CONFIG_PREFIX + "*", MappingSource.INDEX);
+    assertThat(mappingsBeforeOpen.keySet()).isEmpty();
+    final String[] expectedIndices = getExpectedIndices(config);
+
+    // when
+    camundaExporter.open(new ExporterTestController());
+
+    // then
+    final var mappingsAfterOpen =
+        adapter.getSearchEngineClient().getMappings(CONFIG_PREFIX + "*", MappingSource.INDEX);
+    assertThat(mappingsAfterOpen.keySet())
+        .isNotEmpty()
+        .hasSize(expectedIndices.length)
+        // we verify the names hard coded on purpose
+        // to make sure no index will be accidentally dropped, names are changed or added
+        .containsExactlyInAnyOrder(
+            "custom-prefix-identity-authorizations-8.7.0_",
```

Comment:
We already do all the work to calculate the expected indices why not use that as the expected field

Reply:
Because if we remove the descriptor from the DefaultResourceProvider the test will not fail, as it would reuse this. So we have no prevention of mistakenly remove an descriptor, etc.

Reply:
ah ok makes sense